scylla-cql/src/types/serialize/row.rs

```diff
@@ -200,3 +206,127 @@ mod tests {
         assert_eq!(sorted_row_data, unsorted_row_data);
     }
 }
+
+/// Keeps a buffer with serialized Values
+/// Allows adding new Values and iterating over serialized ones
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct NewSerializedValues {
+    serialized_values: Vec<u8>,
+}
+
+impl NewSerializedValues {
+    pub fn new() -> Self {
+        let mut buf = Vec::new();
+        buf.extend_from_slice(&0u16.to_be_bytes());
+        NewSerializedValues {
+            serialized_values: buf,
+        }
+    }
+    pub fn from_serializable<T: SerializeRow>(
+        ctx: &RowSerializationContext,
+        row: &T,
+    ) -> Result<Self, SerializationError> {
+        let mut data = Vec::new();
+        let mut writer = BufBackedRowWriter::new(&mut data);
+        T::preliminary_type_check(ctx)?;
+        row.serialize(ctx, &mut writer)?;
+        drop(writer);
+        Ok(NewSerializedValues {
+            serialized_values: data,
+        })
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
```

Comment:
Actually no, this looks fine - len() returns the element count and size() returns buffer size. I'm still going to rename the methods because it's confusing.

Reply:
I agree that the names are confusing and it makes sense to rename.